docs/apache-airflow/installation/installing-from-pypi.rst

```diff
@@ -79,50 +87,86 @@ where:
 - ``AIRFLOW_VERSION`` - Airflow version (e.g. :subst-code:`|version|`) or ``main``, ``2-0``, for latest development version
 - ``PYTHON_VERSION`` Python version e.g. ``3.8``, ``3.9``
 
-There is also a ``constraints-no-providers`` constraint file, which contains just constraints required to
-install Airflow core. This allows to install and upgrade airflow separately and independently from providers.
+**However, that should not prevent you from being able to upgrade or downgrade providers and dependencies to other versions**
 
-You can create the URL to the file substituting the variables in the template below.
+You can, for example, install new versions of providers and dependencies after the release to use the latest
+version and up-to-date with latest security fixes - even if you do not want upgrade airflow core version.
+Or you can downgrade some dependencies or providers if you want to keep previous versions for compatibility
+reasons. Installing such dependencies should be done without constraints as a separate pip command.
 
-.. code-block::
+When you do such an upgrade, you should make sure to also add the ``apache-airflow`` package to the list of
+packages to install and pin it to the version that you have, otherwise you might end up with a
+different version of Airflow than you expect as ``pip`` can upgrade/downgrade it automatically when
+performing dependency resolution.
 
-  https://raw.githubusercontent.com/apache/airflow/constraints-${AIRFLOW_VERSION}/constraints-no-providers-${PYTHON_VERSION}.txt
+.. code-block:: bash
 
-You can also use "latest" as version when you install "latest" stable version of Airflow. The "latest"
-constraints always points to the "latest" released Airflow version constraints:
+    pip install "apache-airflow[celery]==|version|" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-|version|/constraints-3.8.txt"
+    pip install "apache-airflow[celery]==|version|" apache-airflow-providers-google==10.1.1
```

Comment:
Hmm, is it necessary to specify extras again?

Reply:
Ah... NO :) Copy&paste -> great spot !